flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/config/KinesisSourceConfigOptions.java

```diff
@@ -33,28 +36,35 @@ public enum InitialPosition {
     }
 
     public static final ConfigOption<InitialPosition> STREAM_INITIAL_POSITION =
-            ConfigOptions.key("flink.stream.initpos")
+            ConfigOptions.key("kinesis.stream.init.position")
                     .enumType(InitialPosition.class)
                     .defaultValue(InitialPosition.LATEST)
                     .withDescription("The initial position to start reading Kinesis streams.");
 
     public static final ConfigOption<String> STREAM_INITIAL_TIMESTAMP =
-            ConfigOptions.key("flink.stream.initpos.timestamp")
+            ConfigOptions.key("kinesis.stream.init.timestamp")
                     .stringType()
                     .noDefaultValue()
                     .withDescription(
                             "The initial timestamp at which to start reading from the Kinesis stream. This is used when AT_TIMESTAMP is configured for the STREAM_INITIAL_POSITION.");
 
     public static final ConfigOption<String> STREAM_TIMESTAMP_DATE_FORMAT =
-            ConfigOptions.key("flink.stream.initpos.timestamp.format")
+            ConfigOptions.key("kinesis.stream.init.timestamp.format")
                     .stringType()
                     .defaultValue("yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
                     .withDescription(
                             "The date format used to parse the initial timestamp at which to start reading from the Kinesis stream. This is used when AT_TIMESTAMP is configured for the STREAM_INITIAL_POSITION.");
 
-    public static final ConfigOption<Long> SHARD_DISCOVERY_INTERVAL_MILLIS =
-            ConfigOptions.key("flink.shard.discovery.intervalmillis")
-                    .longType()
-                    .defaultValue(10000L)
+    public static final ConfigOption<Duration> SHARD_DISCOVERY_INTERVAL =
+            ConfigOptions.key("kinesis.shard.discovery.interval")
+                    .durationType()
+                    .defaultValue(Duration.ofSeconds(10))
                     .withDescription("The interval between each attempt to discover new shards.");
+
+    public static final ConfigOption<Integer> SHARD_GET_RECORDS_MAX =
+            ConfigOptions.key("kinesis.shard.get-records.max-record-count")
+                    .intType()
+                    .defaultValue(10000)
+                    .withDescription(
+                            "The maximum number of getRecords attempts if we get a recoverable exception.");
```

Comment:
Why do we need this? Also I think this description might not be correct right - this seems to be used as a "getRecords" limit.

Reply:
The old connector has this as configuration. [https://github.com/apache/flink-connector-aws/blob/main/flink-connector-aws/flink-[…]treaming/connectors/kinesis/config/ConsumerConfigConstants.java](https://github.com/apache/flink-connector-aws/blob/main/flink-connector-aws/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/config/ConsumerConfigConstants.java#L245) 
I will update the description.